pkg/api/annotations.go

```diff
@@ -44,6 +44,13 @@ func (hs *HTTPServer) GetAnnotations(c *models.ReqContext) response.Response {
 		if item.Email != "" {
 			item.AvatarUrl = dtos.GetGravatarUrl(item.Email)
 		}
+		if item.DashboardId != 0 {
+			query := models.GetDashboardQuery{Id: item.DashboardId, OrgId: c.OrgId}
+			err := hs.SQLStore.GetDashboard(c.Req.Context(), &query)
+			if err == nil && query.Result != nil {
+				item.DashboardUID = &query.Result.Uid
+			}
+		}
```

Comment:
This seems too heavy to run for each annotation?

Reply:
nice point, we can build a "cache", call only once per dashboard...